nixos/modules/config/sysctl.nix

```diff
@@ -21,11 +21,19 @@ in
   options = {
 
     boot.kernel.sysctl = mkOption {
+      type = types.submodule {
+        freeformType = types.attrsOf sysctlOption;
+        options."net.core.rmem_max" = mkOption {
+          type = types.ints.unsigned // {
+            merge = loc: defs: foldl (a: b: lib.max a.value b.value) {value = 0;} defs;
+          };
```

Comment:
This doesn't work though, as you cannot compare an integer with null.

Reply:
If my suggestted change is applied intactly, it works.

Reply:
yes, that's what I'm testing right now, I think that is the right solution.
Just making sure.

Reply:
I think the type should be changed to `types.nullOr ...` too, since `null` is not unsigned int.

Reply:
that makes sense, it makes me think that we should change the merge function to accept a null in b.value then.
However it complicates thing without providing any value since, no one will really use null as a value manually.

Reply:
btw, I've just changed the type.

Reply:
`null` is used as a default value to avoid setting `net.core.rmem_max` to `0` (I check on a vps the kernel's default value is 212992, not 0) if `net.core.rmem_max` is not set at all because `null` will be filtered in https://github.com/NixOS/nixpkgs/blob/bc5ad970780580ec272b3a6bc5f9bddf4e250d3c/nixos/modules/config/sysctl.nix#L46.

Reply:
I've changed the merge function to take into account the null. This was causing errors in the CI.
Thank you!